keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,908 @@
+-->
+
+Since kubelet runs as root an attacker can configure a Git repo to execute
+attacker control code on the host node as root. In [CVE-2024-10220](https://irsl.medium.com/sneaky-write-hook-git-clone-to-root-on-k8s-node-e38236205d54)
+security researchers exploited gitRepo volumes and Git hooks to execute attacker
+controlled code on the host node as root.
+
+gitRepo has been deprecated for almost 5 years and the kubernetes 
+[documentation](https://kubernetes.io/docs/concepts/storage/volumes/#gitrepo)
+warns users against using it and provides a VAP policy to block its use. 
+Kubernetes also recommends mounting an emptyDir volume into an init container 
+that clones the repo using Git, then mounting the emptyDir into the Pod's main
+container. The git-sync project (https://github.com/kubernetes/git-sync) is
+designed for exactly this.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Remove in-tree gitRepo volume support.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Kubernetes drops support for gitRepo volume types by removing the in-tree driver
+that actually mounts the volume in the pod. In this case we will still keep the
+gitRepo volume in the API. This removal is similar to removal of support for
+[glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+which was deprecated in v1.25 and removed in v1.26.
+
+Sig-storage was actually very close to doing this and had even added a notice of
+removal in the release notes for v1.32. But due to concerns raised by sig-arch
+they rolled back the notice. sig-storage provided [this rationale](https://github.com/kubernetes/kubernetes/issues/125983#issuecomment-2522201283) for considering removal.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+We will add a new feature-gate called `DisableGitRepoVolumeDriver` which will be
+defaulted to `false`. When this feature-gate is set to `true` kubelet will not
+sync and mount gitRepo volumes and this will disable the [gitRepo driver](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/git_repo/git_repo.go).
+
+### Timeline:
+v1.33 (04/2025): 
```

Comment:
In an unusual twist, I want to argue to accelerate and skip this, jumping to step 2.  We need to convince a few folks, I am sure.  @liggitt  @deads2k perhaps?

Reply:
since there's a security hole that doesn't get closed by default until step 2, and there's no risk of accelerating to step 2 (we're just disabling old code faster, not enabling new untested code faster), I'd agree

Reply:
Updated, to a more "aggressive" timeline.